lib/propolis/src/block/crucible.rs

```diff
@@ -332,28 +338,50 @@ impl From<Error> for block::Result {
 
 async fn process_request(
     block: &(dyn BlockIO + Send + Sync),
-    read_only: bool,
+    info: &block::DeviceInfo,
     skip_flush: bool,
     req: &block::Request,
+    readbuf: &mut Buffer,
     mem: &MemCtx,
 ) -> Result<(), Error> {
+    let block_size = info.block_size as usize;
+
+    // Calculate offset (in crucible::Block form) and length in blocksize
+    fn block_offset_count(
+        off_bytes: usize,
+        len_bytes: usize,
+        block_size: usize,
+    ) -> Result<(crucible::Block, usize), Error> {
+        if off_bytes % block_size == 0 || len_bytes % block_size == 0 {
```

Comment:
This should be `&&` not `||`

Reply:
Perhaps this should be a static member function, so it could be tested?

Reply:
> This should be `&&` not `||`

Gah!  I'd flipped the check, but forgot the operator.  Thanks for catching that.

> Perhaps this should be a static member function, so it could be tested?

Sure, will do that and add a basic test or two